bluemira/base/reactor_config.py

```diff
@@ -234,21 +236,56 @@ def _check_args_are_strings(self, args: Iterable[str]):
             if not isinstance(a, str):
                 raise ReactorConfigError("args must be strings")
 
+    def _get_nested_filepaths(
+        self,
+        current_layer: str,
+        current_arg_key: str,
+        arg_keys: Tuple[str],
+        next_idx: int,
+    ) -> Dict:
+        if not Path(current_layer).resolve().is_file():
+            if isinstance(self._config_path, str):
+                current_layer = str(Path(self._config_path).parent / current_layer)
+            if not Path(current_layer).resolve().is_file():
+                raise FileNotFoundError(f"Cannot find file {current_layer}")
+
+        new_layer = self._read_json_file(current_layer)
+
+        # Dont need to load files repeatedly so put it back in config_data
+        if next_idx == 1:
+            self.config_data[current_arg_key] = new_layer
+        else:
+            tmp_dict = self.config_data
+            for key in arg_keys[: next_idx - 1]:
+                tmp_dict = tmp_dict[key]
+            tmp_dict[current_arg_key] = new_layer
+        return new_layer
+
     def _extract(self, arg_keys: Tuple[str], is_config: bool) -> dict:
         extracted = {}
 
         # this routine is designed not to copy any dict's while parsing
 
         current_layer = self.config_data
+
         for next_idx, current_arg_key in enumerate(arg_keys, start=1):
             current_layer = current_layer.get(current_arg_key, {})
             next_arg_key = arg_keys[next_idx] if next_idx < len(arg_keys) else None
 
+            if isinstance(current_layer, str) and current_layer.endswith(".json"):
```

Comment:
I think we need to chat about the scope of using nested file paths.
You might have some keys with file path values, that need to be expanded that are in one of args in `.config_for`, `.params_for` etc.


I also never thought about the case of current_layer being a `str`, which could happen if one the args drills down to a string.